static/conformance-view.js

```diff
@@ -0,0 +1,314 @@
+
+        var compilationOptions = $("<input>")
+            .attr("class", "options form-control")
+            .attr("type", "text")
+            .attr("size", "256")
+            .attr("placeholder", "Compiler options...")
+            .attr("data-cv", config.cv)
+            .val(config.options);
+
+        this.selectorList.append($("<tr>")
+            .attr("data-cv", config.cv)
+            .append($("<td>")
+                .append(status)
+            ).append($("<td>")
+                .append(newSelector)
+            ).append($("<td>")
+                .append(compilationOptions)
+            ).append($("<td>")
+                .append($("<button>")
+                    .attr("class", "close")
+                    .attr("aria-label", "Close")
+                    .attr("data-cv", config.cv)
+                    .append($("<span>")
+                        .html("&times;")
+                        .attr("aria-hidden", "true")
+                    )
+                    .on("click", _.bind(function() {
+                        this.removeCompilerSelector(config.cv);
+                    }, this))
+                )
+            )
+        );
+
+        newSelector.selectize({
+            sortField: 'name',
+            valueField: 'id',
+            labelField: 'name',
+            searchField: ['name'],
+            options: compilers,
+            items: config.compilerId ? [config.compilerId] : []
+        }).on('change', _.bind(function() {
+            // Hide the results button when a new compiler is selected
+            this.handleStatusIcon(status, {code: 0, text: ""});
+            this.saveState();
+        }, this));
+
+        this.handleStatusIcon(status, config.status);
+        this.handleToolbarUI();
+    };
+
+    Conformance.prototype.removeCompilerSelector = function(cv) {
+        _.each(this.selectorList.children(), function(row) {
+            var child = $(row);
+            if (child.attr("data-cv") == cv) {
+                child.remove();
+            }
+        }, this);
+        this.handleToolbarUI();
+        this.saveState();
+    };
+
+    // TODO: Maybe use premises
+    Conformance.prototype.compile = function(selector) {
+        if (!selector) return;
+        var val = selector.val();
+        if (val) {
+            var cv = Number(selector.attr("data-cv"));
+            var request = {
+                source: this.currentExpandedSource || "",
+                compiler: val,
+                options: this.selectorList.find(".options[data-cv='" + cv + "']").val(),
+                backendOptions: {produceAst: false},
+                filters: {},
+                extras: {
+                    cv: cv,
+                    storeAsm: false,
+                    emitCompilingEvent: false,
+                    ignorePendingRequest: true
+                }
+            };
+            Compiler.Compiler.prototype.sendCompile(request, _.bind(this.onCompileResponse, this));
```

Comment:
That was my intention. That this would point here so I wouldn't need to pass the onCompileResponse. But the final solution is a mix of both. So I'll just do what you say and extract those functions into their own file. Any suggestion for said file's name?